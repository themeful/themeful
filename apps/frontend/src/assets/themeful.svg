<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 2000 2000">
  <path fill="#333" d="M1166.667 0 833.333 2000H200c-110.457 0-200-89.543-200-200V200C0 89.543 89.543 0 200 0h966.667Z"/>
  <path fill="#EEE" d="M1141.667 150 858.333 1850h-659.16c-26.96-.444-48.734-22.22-49.166-49.181L150 1800V199.173C150.448 171.94 172.662 150 200 150h941.667Z"/>
  <path fill="#B21CE2" d="M596.602 1637.336c-90.235 0-160.547-25.635-210.938-76.904-50.39-51.27-75.586-132.276-75.586-243.018V773.371H198.457v-137.11h111.621V354.134h165.235v282.129h304.101v137.11H475.313v466.698c0 49.805.878 88.184 2.636 115.137 1.758 26.953 8.496 52.148 20.215 75.586 9.961 20.508 26.074 35.889 48.34 46.143 22.266 10.253 52.441 15.38 90.527 15.38 26.953 0 53.028-3.955 78.223-11.865s43.36-14.502 54.492-19.775h9.668v148.535a798.454 798.454 0 0 1-94.482 20.654c-31.934 4.98-61.377 7.47-88.33 7.47Z"/>
  <path fill="#EEE" d="M1800 0c110.457 0 200 89.543 200 200v1600c0 110.457-89.543 200-200 200H833.334L1166.667 0H1800Z"/>
  <path fill="#333" d="M1800.827 150c27.233.448 49.173 22.662 49.173 50v1600.827c-.448 27.233-22.662 49.173-50 49.173H858.333l283.333-1700Z"/>
  <path fill="#2EF7FF" d="M1390.898 876.203v529.102c0 75 8.204 122.46 24.61 142.382 21.68 25.782 50.683 38.672 87.012 38.672h72.949V1618h-480.762v-31.64h36.035c23.438 0 44.824-5.86 64.16-17.579 19.336-11.718 32.666-27.539 39.99-47.46 7.325-19.923 10.987-58.594 10.987-116.016V876.203h-156.445v-63.281h156.445v-52.735c0-80.273 12.89-148.242 38.672-203.906 25.781-55.664 65.185-100.635 118.213-134.912 53.027-34.277 112.646-51.416 178.857-51.416 61.524 0 118.066 19.922 169.629 59.766 33.984 26.367 50.977 55.957 50.977 88.77 0 17.577-7.618 34.13-22.852 49.657-15.234 15.528-31.64 23.292-49.219 23.292-13.476 0-27.685-4.834-42.627-14.502-14.941-9.668-33.252-30.47-54.931-62.403-21.68-31.933-41.602-53.467-59.766-64.6-18.164-11.132-38.379-16.699-60.645-16.699-26.953 0-49.804 7.178-68.554 21.534-18.75 14.355-32.227 36.62-40.43 66.796-8.203 30.176-12.305 107.96-12.305 233.35v58.008h207.422v63.281h-207.422Z"/>
</svg>
