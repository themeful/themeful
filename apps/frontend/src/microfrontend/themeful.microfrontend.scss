:host {
  display: block;
  color: var(--tf-font-color);
  font: var(--tf-default-font);
}

header {
  position: fixed;
  width: 100%;
  background-color: var(--tf-content-bg-color);
  z-index: 1;
  display: flex;
  flex-direction: row;
  padding: 8px 8px 0 8px;
}

.navigation {
  width: 100%;
}

nav {
  display: flex;
  justify-content: left;
  background-color: var(--tf-content-bg-color);
}

main {
  padding-top: 90px;
}

h1 {
  line-height: 1.5rem;
  font-size: 20px;
  margin-top: 0;
  margin-bottom: 4px;
}

.nav-row {
  display: block;
}

.logo {
  width: 56px;
  height: 56px;
  margin-right: 8px;
}

.mode-toggle {
  float: right;
}
