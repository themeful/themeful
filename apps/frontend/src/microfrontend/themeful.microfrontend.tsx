import { Component, h, Host, State } from '@stencil/core'
import { APIService } from '@themeful-ui'

@Component({
  tag: 'themeful-microfrontend',
  styleUrl: 'themeful.microfrontend.scss',
  shadow: true,
})
export class ThemefulMicrofrontend {
  private apiService: APIService
  private styleGuides$
  private bundle$
  private toast$
  @State() mode: string

  constructor() {
    this.apiService = APIService.Instance
    this.styleGuides$ = this.apiService.styleGuides
    this.bundle$ = this.apiService.bundle
    this.toast$ = this.apiService.toast
  }

  private onAction = ({ detail }) => {
    if (!['open', 'close'].includes(detail.action)) {
      this.apiService.action(detail).subscribe((result) => {
        this.toast$.next({
          text: result ? 'Updated' : 'Something went wrong',
          status: result ? 'success' : 'error',
        })
      })
    }
  }

  private toogleMode = () => {
    document.body.classList.toggle('tf-light')
    this.checkMode()
  }

  private checkMode = () => {
    this.mode = document.body.classList.contains('tf-light') ? 'light' : 'dark'
  }

  public componentWillLoad(): void {
    this.checkMode()
  }

  public render(): HTMLThemefulMicrofrontendElement {
    return (
      <Host>
        <header>
          <div class="logo-bar">
            <img class="logo" src="/assets/themeful.svg" />
            <h1>Themeful</h1>
          </div>
          <nav class="button-group nav-row">
            <stencil-route-link url="/themes">
              <tf-button kind="selectable">Themes</tf-button>
            </stencil-route-link>
            <stencil-route-link url="/styleguides">
              <tf-button kind="selectable">Style Guides</tf-button>
            </stencil-route-link>
            <tf-button class="mode-toggle" onClick={this.toogleMode} kind="selectable">
              {this.mode === 'light' ? 'Dark' : 'Light'} Mode
            </tf-button>
          </nav>
        </header>
        <tf-toast {...{ msg$: this.toast$ }} />
        <stencil-router>
          <stencil-route-switch scrollTopOffset={0}>
            <stencil-route
              url="/themes"
              component="tf-themes"
              componentProps={{
                bundle$: this.bundle$,
                onAction: this.onAction,
              }}
            />
            <stencil-route
              url="/styleguides"
              component="tf-style-guides"
              componentProps={{
                styleGuides$: this.styleGuides$,
                onAction: this.onAction,
              }}
            />
            <stencil-route
              url="/styleguide/:slug"
              component="tf-style-guide-details"
              componentProps={{
                styleGuides$: this.styleGuides$,
                onAction: this.onAction,
              }}
            />
            <stencil-route url="/" exact={true}>
              <stencil-router-redirect url="/themes" />
            </stencil-route>
          </stencil-route-switch>
        </stencil-router>
      </Host>
    )
  }
}
