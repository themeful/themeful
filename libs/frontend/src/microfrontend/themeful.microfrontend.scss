:host {
  display: block;
  color: var(--tf-font-color);
  font: var(--tf-default-font);
}
