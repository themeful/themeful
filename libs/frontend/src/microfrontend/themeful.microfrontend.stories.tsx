import { h } from '@stencil/core'

export default {
  title: 'Microfrontend/Themeful',
}

export const themeful = () => {
  return <themeful-microfrontend />
}
