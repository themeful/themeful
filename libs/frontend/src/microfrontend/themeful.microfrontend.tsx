import { Component, h } from '@stencil/core'
import { APIService } from '../services/'

@Component({
  tag: 'themeful-component',
  styleUrl: 'themeful.microfrontend.scss',
  shadow: true,
})
export class ThemefulComponent {
  private apiService: APIService

  constructor() {
    this.apiService = APIService.Instance
  }

  private getText(): string {
    return this.apiService.getHello()
  }

  render() {
    return <div>Hello, World! {this.getText()}</div>
  }
}
