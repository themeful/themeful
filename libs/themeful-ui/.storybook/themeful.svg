<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 3021 630">
  <g fill="none" fill-rule="evenodd">
    <path fill="#D546C5" d="m41 121 85-14v111h130v71H126v149c0 29 5 50 14 63 9 12 25 19 48 19 15 0 29-2 41-5l28-9 14 67-39 13c-16 4-34 6-56 6-26 0-48-3-65-10-18-7-31-17-42-31-10-13-17-29-21-48-5-19-7-40-7-65V121Zm297 463V55l85-14v181a226 226 0 0 1 69-12c29 0 54 4 73 12 20 9 35 20 47 35 11 14 20 32 25 53 5 20 7 43 7 68v206h-85V392c0-20-1-37-4-51-2-13-6-25-12-33-6-9-14-15-24-19-9-4-22-6-36-6a235 235 0 0 0-60 10v291h-85Zm387-181c0-33 5-61 15-85 9-24 22-44 38-61a169 169 0 0 1 118-49c50 1 89 16 117 48 28 31 43 78 43 140a398 398 0 0 1-2 31H813c2 30 12 52 31 68 18 17 45 25 80 25a264 264 0 0 0 97-18l11 69-19 8a272 272 0 0 1-62 12l-38 2c-32 0-60-5-84-14a156 156 0 0 1-93-100c-7-23-11-49-11-76Zm246-38c0-12-2-23-5-33-3-11-8-20-14-28a68 68 0 0 0-56-24 76 76 0 0 0-59 26c-7 8-12 17-16 27-4 11-6 21-8 32h158Zm383 27c0-39-5-66-15-83-9-17-27-26-53-26a250 250 0 0 0-56 6v295h-85V230a565 565 0 0 1 148-20c25 0 46 3 62 10s29 15 40 26a165 165 0 0 1 46-24 243 243 0 0 1 65-12c28 0 52 4 70 12 18 9 33 20 43 35 11 14 18 32 22 53s6 43 6 68v206h-85V392c0-39-4-66-14-83-9-17-27-26-54-26-13 0-26 2-38 7-12 4-21 8-27 13a395 395 0 0 1 10 79l-1 202h-84V392Zm376 11c0-33 5-61 15-85 9-24 22-44 38-61a169 169 0 0 1 118-49c50 1 89 16 117 48 29 31 43 78 43 140a398 398 0 0 1-2 31h-241c2 30 13 52 31 68 18 17 45 25 80 25a264 264 0 0 0 97-18l11 69-19 8a272 272 0 0 1-62 12l-37 2c-33 0-61-5-84-14a156 156 0 0 1-93-100c-8-23-12-49-12-76Zm246-38c0-12-2-23-5-33-3-11-8-20-14-28a68 68 0 0 0-56-24 76 76 0 0 0-59 26c-7 8-12 17-16 27-3 11-6 21-7 32h157Zm323-324a228 228 0 0 1 86 15l-16 70c-8-4-17-7-28-10-10-2-22-3-34-3-13 0-25 2-34 6a56 56 0 0 0-34 42c-3 10-4 21-4 32v25h133v71h-133v295h-85V192c0-47 13-84 37-111 25-27 63-40 112-40Zm432 532-64 13c-26 4-55 6-86 6-30 0-54-4-74-12-20-9-36-20-48-35s-20-33-25-54-8-43-8-68V218h85v192c0 39 6 67 17 84s31 25 60 25a401 401 0 0 0 58-5V218h85v355Zm235 18c-25 0-46-3-63-8-16-6-30-13-39-22a78 78 0 0 1-21-36c-4-14-6-31-6-49V55l85-14v419l2 26c2 7 5 13 9 18s10 8 17 11 17 5 28 6l-12 70Z"/>
    <path fill="#4671D5" d="M126 441c0 28 5 48 14 60s25 19 48 19c15 0 29-2 41-5l28-9 14 67-39 13c-16 4-34 6-56 6-26 0-48-3-65-10-18-7-31-17-42-31-10-13-17-29-21-48-4-17-6-36-7-58l85-4Zm212-11 85-4v158h-85V430Zm306-16v170h-85V418l85-4Zm412-22a404 404 0 0 1-2 35H813c2 30 12 52 31 68 18 17 45 25 80 25a264 264 0 0 0 97-18l11 69-19 8a272 272 0 0 1-62 12l-38 2c-32 0-60-5-84-14a156 156 0 0 1-93-100c-6-21-10-44-11-69l331-18Zm297-15 85-5v212h-84V392l-1-15Zm-123 6v201h-85V388l85-5Zm417-22v223h-85V366l85-5Zm87-4 87-5-2 13h157l-2-21 82-4a321 321 0 0 1 3 87h-241c2 30 13 52 31 68 18 17 45 25 80 25a264 264 0 0 0 97-18l11 69-19 8a272 272 0 0 1-62 12l-37 2c-33 0-61-5-84-14a156 156 0 0 1-93-100 246 246 0 0 1-8-122Zm501-26v253h-85V335l85-4Zm191-10 85-5v94c0 39 6 67 17 84s31 25 60 25a401 401 0 0 0 58-5V309l85-4v268l-64 13c-26 4-55 6-86 6-30 0-54-4-74-12-20-9-36-20-48-35s-20-33-25-54-8-43-8-68V321Zm411-22 85-4v165l2 26c2 7 5 13 9 18s10 8 17 11 17 5 28 6l-12 70c-25 0-46-3-63-8-16-6-30-13-39-22a78 78 0 0 1-21-36c-4-14-6-31-6-49V299Z"/>
  </g>
</svg>
