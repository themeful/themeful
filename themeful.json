{
  "paths": {
    "generatedPath": "./sample/generated/",
    "dataPath": "./sample/generated/",
    "themesPath": "./sample/generated/",
    "libPaths": ["./sample/components/button/", "./sample/components/card"]
  },
  "global": {
    "baseFontSize": 16,
    "shortDesignTokens": false
  }
}
