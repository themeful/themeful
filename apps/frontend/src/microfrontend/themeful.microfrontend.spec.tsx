import { newSpecPage } from '@stencil/core/testing'
import { ThemefulMicrofrontend } from './themeful.microfrontend'

describe('ThemefulMicrofrontend', () => {
  it('renders', async () => {
    const { root } = await newSpecPage({
      components: [ThemefulMicrofrontend],
      html: '<themeful-microfrontend></themeful-microfrontend>',
    })
    expect(root).toEqualHtml(`
      <themeful-microfrontend>
        <mock:shadow-root>
          <header>
            <stencil-route-link url="/themes">
              <img class="logo" src="/assets/themeful.svg">
            </stencil-route-link>
            <div class="navigation">
              <h1>
                Themeful
              </h1>
              <nav class="button-group nav-row">
                <stencil-route-link url="/themes">
                  <tf-button kind="selectable">
                    Themes
                  </tf-button>
                </stencil-route-link>
                <stencil-route-link url="/styleguides">
                  <tf-button kind="selectable">
                    Style Guides
                  </tf-button>
                </stencil-route-link>
                <stencil-route-link url="/components">
                  <tf-button kind="selectable">
                    Components
                  </tf-button>
                </stencil-route-link>
                <tf-button class="mode-toggle" kind="selectable">
                  Light Mode
                </tf-button>
              </nav>
            </div>
            <tf-toast></tf-toast>
          </header>
          <main>
            <stencil-router>
              <stencil-route-switch scrolltopoffset="0">
                <stencil-route component="tf-themes" url="/themes"></stencil-route>
                <stencil-route component="tf-style-guides" url="/styleguides"></stencil-route>
                <stencil-route component="tf-style-guide-details" url="/styleguide/:slug"></stencil-route>
                <stencil-route component="tf-components" url="/components"></stencil-route>
                <stencil-route exact="" url="/">
                  <stencil-router-redirect url="/themes"></stencil-router-redirect>
                </stencil-route>
              </stencil-route-switch>
            </stencil-router>
          </main>
        </mock:shadow-root>
      </themeful-microfrontend>
    `)
  })

  it('renders with values', async () => {
    const { root } = await newSpecPage({
      components: [ThemefulMicrofrontend],
      html: `<themeful-microfrontend></themeful-microfrontend>`,
    })
    expect(root).toEqualHtml(`
      <themeful-microfrontend>
        <mock:shadow-root>
        <header>
          <stencil-route-link url="/themes">
            <img class="logo" src="/assets/themeful.svg">
          </stencil-route-link>
          <div class="navigation">
            <h1>
              Themeful
            </h1>
            <nav class="button-group nav-row">
              <stencil-route-link url="/themes">
                <tf-button kind="selectable">
                  Themes
                </tf-button>
              </stencil-route-link>
              <stencil-route-link url="/styleguides">
                <tf-button kind="selectable">
                  Style Guides
                </tf-button>
              </stencil-route-link>
              <stencil-route-link url="/components">
                <tf-button kind="selectable">
                  Components
                </tf-button>
              </stencil-route-link>
              <tf-button class="mode-toggle" kind="selectable">
                Light Mode
              </tf-button>
            </nav>
          </div>
          <tf-toast></tf-toast>
        </header>
        <main>
          <stencil-router>
            <stencil-route-switch scrolltopoffset="0">
              <stencil-route component="tf-themes" url="/themes"></stencil-route>
              <stencil-route component="tf-style-guides" url="/styleguides"></stencil-route>
              <stencil-route component="tf-style-guide-details" url="/styleguide/:slug"></stencil-route>
              <stencil-route component="tf-components" url="/components"></stencil-route>
              <stencil-route exact="" url="/">
                <stencil-router-redirect url="/themes"></stencil-router-redirect>
              </stencil-route>
            </stencil-route-switch>
          </stencil-router>
        </main>
        </mock:shadow-root>
      </themeful-microfrontend>
    `)
  })
})
