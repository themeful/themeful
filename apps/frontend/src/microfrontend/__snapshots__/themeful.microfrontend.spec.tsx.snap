// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemefulMicrofrontend renders 1`] = `
<themeful-microfrontend>
  <mock:shadow-root>
    <header>
      <a href="/themes">
        <img class="logo" src="/assets/themeful.svg">
      </a>
      <div class="navigation">
        <h1>
          Themeful
        </h1>
        <nav class="button-group nav-row">
          <a href="/themes">
            <tf-button kind="selectable">
              Themes
            </tf-button>
          </a>
          <a href="/styleguides">
            <tf-button kind="selectable">
              Style Guides
            </tf-button>
          </a>
          <a href="/components">
            <tf-button kind="selectable">
              Components
            </tf-button>
          </a>
          <tf-button class="mode-toggle" kind="selectable">
            Light Mode
          </tf-button>
        </nav>
      </div>
      <tf-toast></tf-toast>
    </header>
    <main>
      <tf-themes></tf-themes>
    </main>
  </mock:shadow-root>
</themeful-microfrontend>
`;

exports[`ThemefulMicrofrontend renders with values 1`] = `
<themeful-microfrontend>
  <mock:shadow-root>
    <header>
      <a href="/themes">
        <img class="logo" src="/assets/themeful.svg">
      </a>
      <div class="navigation">
        <h1>
          Themeful
        </h1>
        <nav class="button-group nav-row">
          <a href="/themes">
            <tf-button kind="selectable">
              Themes
            </tf-button>
          </a>
          <a href="/styleguides">
            <tf-button kind="selectable">
              Style Guides
            </tf-button>
          </a>
          <a href="/components">
            <tf-button kind="selectable">
              Components
            </tf-button>
          </a>
          <tf-button class="mode-toggle" kind="selectable">
            Light Mode
          </tf-button>
        </nav>
      </div>
      <tf-toast></tf-toast>
    </header>
    <main>
      <tf-themes></tf-themes>
    </main>
  </mock:shadow-root>
</themeful-microfrontend>
`;
